cpp/src/arrow/compute/kernels/hash_aggregate.cc

```diff
@@ -621,6 +622,11 @@ struct GroupedValueTraits {
 
   static CType Get(const CType* values, uint32_t g) { return values[g]; }
   static void Set(CType* values, uint32_t g, CType v) { values[g] = v; }
+  static Status AppendBuffers(TypedBufferBuilder<CType>& destination, const CType* values,
```

Comment:
nit: prefer raw pointers over mutable references (`TypedBufferBuilder<CType>*`), while the style guide we use no longer requires this, our codebase was written with this rule in place so let's be consistent